backend-kotlin/src/main/kotlin/com/example/backendkotlin/presentation/VillageGrpcService.kt

```diff
@@ -0,0 +1,90 @@
+package com.example.backendkotlin.presentation
+
+import com.example.backendkotlin.domain.Village
+import com.example.backendkotlin.domain.VillageId
+import com.example.backendkotlin.generated.grpc.*
+import com.example.backendkotlin.usecase.CreateVillageUseCase
+import com.example.backendkotlin.usecase.ListVillagesUseCase
+import io.grpc.stub.StreamObserver
+import net.devh.boot.grpc.server.service.GrpcService
+import java.util.*
+
+/**
+ * 村に関するgRPCサービス
+ */
+@GrpcService
+class VillageGrpcService(
+    private val listVillagesUseCase: ListVillagesUseCase,
+    private val createVillageUseCase: CreateVillageUseCase
+) : VillageServiceGrpc.VillageServiceImplBase() {
+    /**
+     * 村を作成する
+     *
+     * @param request 村作成リクエスト
+     * @param responseObserver レスポンス
+     *
+     * @return 村
+     */
+    override fun createVillage(
+        request: CreateVillageRequest,
+        responseObserver: StreamObserver<CreateVillageResponse>,
+    ) {
+        val newVillageId = VillageId(UUID.randomUUID())
+        val newVillageRequest = Village(
+            id = newVillageId,
+            name = request.name,
+            citizenCount = request.citizenCount,
+            werewolfCount = request.werewolfCount,
+            fortuneTellerCount = request.fortuneTellerCount,
+            knightCount = request.knightCount,
+            psychicCount = request.psychicCount,
+            madmanCount = request.madmanCount,
+            isInitialActionActive = request.isInitialActionActive
+        )
```

Comment:
[FYI] 細かいテクニックだけど、scope 関数を使うと何を対象にしているかが明確になって良さげだったり
```suggestion
        val newVillageRequest = request.let { r ->
         Village(
            id = newVillageId,
            name = r.name,
            citizenCount = r.citizenCount,
            werewolfCount = r.werewolfCount,
            fortuneTellerCount = r.fortuneTellerCount,
            knightCount = r.knightCount,
            psychicCount = r.psychicCount,
            madmanCount = request.madmanCount,
            isInitialActionActive = r.isInitialActionActive
        )
   }
```

Reply:
d3fdc27 で修正